ow_faults/include/ow_faults/FaultInjector.h

```diff
@@ -58,97 +53,122 @@ class FaultInjector
-
-  //checking faults
-  void checkArmFaults();
-  void checkAntFaults();
-
   // Find an item in an std::vector or other find-able data structure, and
   // return its index. Return -1 if not found.
   template<typename group_t, typename item_t>
   int findPositionInGroup(const group_t& group, const item_t& item);
-
   // Get index from m_joint_index_map. If found, modify out_index and return
   // true. Otherwise, return false.
   bool findJointIndex(const unsigned int joint, unsigned int& out_index);
 
-  ow_faults::FaultsConfig m_faults;
+  // power functions
+  float getRandomFloatFromRange(float min_val, float max_val);
+  void publishPowerSystemFault();
+  void powerSOCListener(const std_msgs::Float64& msg);
+  void powerTemperatureListener(const std_msgs::Float64& msg);
 
-  //component faults
-  bool m_armFault;
-  bool m_antFault;
+  // Antennae functions
+  void antennaePanFaultCb(const std_msgs::Float64& msg);
+  void antennaeTiltFaultCb(const std_msgs::Float64& msg);
+  void publishAntennaeFaults(const std_msgs::Float64& msg, bool encoder, 
+                             bool torque, float& m_faultValue, ros::Publisher& m_publisher);
 
+  //Setting message values
+  template<typename fault_msg>
+  void setFaultsMessageHeader(fault_msg& msg);
+  template<typename bitsetFaultsMsg, typename bitmask>
+  void setBitsetFaultsMessage(bitsetFaultsMsg& msg, bitmask systemFaultsBitmask);
+  template<typename fault_msg>
+  void setComponentFaultsMessage(fault_msg& msg, ComponentFaults value);
+ 
+  //checking rqt faults
+  void checkArmFaults();
+  void checkAntFaults();
+
+  ///////publishers and subsscribers
   // arm faults
   ros::Subscriber m_joint_state_sub;
   ros::Publisher m_joint_state_pub;
 
+  //power
+  ros::Subscriber m_power_soc_sub;
+  ros::Subscriber m_power_temperature_sub;
+  ros::Publisher m_power_fault_trigger_pub;
+
   // ft sensor
   ros::Subscriber m_dist_pitch_ft_sensor_sub;
   ros::Publisher m_dist_pitch_ft_sensor_pub;
 
-  // temporary placeholder publishers until power feautre is finished
-  ros::Publisher m_fault_power_state_of_charge_pub;
-  ros::Publisher m_fault_power_temp_pub;
-
-  // publishers for sending ros messages for component failures
-  ros::Publisher m_fault_status_pub;
-  ros::Publisher m_arm_fault_status_pub;
-  ros::Publisher m_power_fault_status_pub;
-  ros::Publisher m_antennae_fault_status_pub;
-
-  //antenna pub and subs
+  //antenna 
   ros::Subscriber m_fault_ant_pan_sub;
   ros::Subscriber m_fault_ant_tilt_sub;
-  ros::Publisher m_fault_ant_pan_pub;
-  ros::Publisher m_fault_ant_tilt_pub;
+  ros::Publisher m_fault_ant_pan_remapped_pub;
+  ros::Publisher m_fault_ant_tilt_remapped_pub;
+
+  // jpl message publishers
+  ros::Publisher m_system_fault_jpl_msg_pub;
+  ros::Publisher m_arm_fault_jpl_msg_pub;
+  ros::Publisher m_power_fault_jpl_msg_pub;
+  ros::Publisher m_antennae_fault_jpl_msg_pub;
+
+  ////////// vars
+  //system
+  std::bitset<10> systemFaultsBitset;
```

Comment:
please follow [OceanWATERS member variable naming guidelines](https://babelfish.arc.nasa.gov/confluence/display/OCEANWATERS/Coding+Style+and+Practices)